File: dialogio/Dialog_ShowItem.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

static void Dialog_ShowItem (void)
{
  if (_dlg.idx2 != 0)
  {
    if (_dlg.idx2 != _dlg.idx3)
      ShowCStr (screen_ptr,
                _dlg.dbuf[_dlg.idx3 - 1].pos,
                _dlg.ystart + _dlg.num + 1,
                (String *)&_dlg.dbuf[_dlg.idx3 - 1].str,
                dl_setting.keys_attr,
                dl_setting.short_attr);

    ShowCStr (screen_ptr,
              _dlg.dbuf[_dlg.idx2 - 1].pos,
              _dlg.ystart + _dlg.num + 1,
              (String *)&_dlg.dbuf[_dlg.idx2 - 1].str,
              dl_setting.keys2_attr,
              dl_setting.short2_attr);

    _dlg.idx3 = _dlg.idx2;
  }

  //EXIT //Dialog_ShowItem
}

File: dialogio/pas/Dialog_ShowItem.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

// Local to `Dialog' procedure

procedure ShowItem;
begin
  If (idx2 = 0) then
    EXIT; //Dialog.ShowItem
  If (idx2 <> idx3) then
    ShowCStr(screen_ptr,dbuf[idx3].pos,ystart+num+1,dbuf[idx3].str,
             dl_setting.keys_attr,dl_setting.short_attr);

    ShowCStr(screen_ptr,dbuf[idx2].pos,ystart+num+1,dbuf[idx2].str,
             dl_setting.keys2_attr,dl_setting.short2_attr);
  idx3 := idx2;

  //EXIT //Dialog.ShowItem
end;
